mne/utils/mixin.py

```diff
@@ -209,8 +209,23 @@ def _getitem(
             key_selection = inst.selection[select]
             drop_log = list(inst.drop_log)
             if reason is not None:
-                for k in np.setdiff1d(inst.selection, key_selection):
-                    drop_log[k] = (reason,)
+                # Used for multiple reasons
+                if isinstance(reason, tuple):
+                    reason = list(reason)
+
+                if isinstance(reason, list):
+                    for i, idx in enumerate(
+                        np.setdiff1d(inst.selection, key_selection)
+                    ):
+                        r = reason[i]
+                        if isinstance(r, str):
+                            r = (r,)
+                        if isinstance(r, list):
+                            r = tuple(r)
+                        drop_log[idx] = r
+                else:
+                    for idx in np.setdiff1d(inst.selection, key_selection):
+                        drop_log[idx] = (reason,)
```

Comment:
Docstring needs to be updated above as well to mention that `reason` can be `list or tuple of str` as well. I don't think we actually want to support `reason` being a `tuple of tuple of str`, like the logic here seems to indicate. Let's just support `reason` being a `list or tuple of str` for which *all unselected epochs get the same `reason`*, i.e.,:
```suggestion
                _validate_type(reason, (list, tuple, str), "reason")
                if isinstance(reason, str):
                    reason = (reason,)
                reason = tuple(reason)
                for i, idx in enumerate(
                    np.setdiff1d(inst.selection, key_selection)
                ):
                    drop_log[idx] = reason
```